packages/cli/src/error_handler.ts

```diff
@@ -5,6 +5,9 @@ import { extractSubCommands } from './extract_sub_commands.js';
 
 let hasAttachUnhandledExceptionListenersBeenCalled = false;
 
+//Create a Map to store error handling state
+const handledErrors = new Map<Error, boolean>();
```

Comment:
why not just use `let hasError = false` and then change it to `true`?

Reply:
got it!

Reply:
Also, if you only use `handledErrors` in `handleError`, it should be defined inside the function's scope. Even if you define a varible shared between several functions, it should be defined close to there it's used.